### ISSUE TITLE
A 100%-chance hanging scenario

### ISSUE THREAD

**Issue by chainikdn** [CrendKing/avisynth_filter#14]
Open video, wait for SVP to kick-in and stabilize, move the quality/perf slider in SVP and immediately after that click on the video timeline. There's probably a race condition when seeking while the filter is still "working" on script reloading via RC, IF script reloading takes a lot of time (such as SVP script with GPU acceleration enabled). 
I can't trigger hang with seeking only or moving SVP's slider only. It only appears when both events happen at the same time.

**Comment by chainikdn** [comment]
BTW I can see that the script is recreated TWICE on every seek. Why?

**Comment by CrendKing** [comment]
I can reproduce the freeze only on madVR. Can you do it on say EVR?

> BTW I can see that the script is recreated TWICE on every seek. Why?

What do you mean? Where do you see it twice?

**Comment by chainikdn** [comment]
> What do you mean? Where do you see it twice?
```
T  19504 @    35271: Start processing output frame   1479 at  308700396 ~  308910395 frameTime     209999 for source    739 Output queue size 32 Front   1480 Back   1511
T   8044 @    35271: Frame handler begin flush
T   8044 @    35271: Frame handler wait for barriers
T  19504 @    35274: GarbageCollect frame    739 pre refcount    1 post queue size 18
T  19504 @    35274: Output worker thread wait for flush
T   8044 @    35398: Frame handler end flush
T   8044 @    35398: ReloadAviSynthScript
T   8044 @    35908: Frame handler begin flush
T   8044 @    35908: Frame handler wait for barriers
T  19504 @    35908: Output worker thread wait for flush
T   8044 @    35912: Frame handler end flush
T   8044 @    35912: ReloadAviSynthScript
T  24400 @    36461: Processed source frame:      0 at          0 ~     417083, nextSourceFrameNb      0 nextOutputFrameStartTime          0
T  24400 @    36465: Processed source frame:      1 at     420000 ~     837083, nextSourceFrameNb      1 nextOutputFrameStartTime          0
T  24400 @    36465: Create output frame      0 for source frame      0 at          0 ~     209999
```

**Comment by chainikdn** [comment]
>  Can you do it on say EVR?

Hmm... nope. But I can't do it with madVR and "GPU accel" off in SVP either. So probably it actually hangs somewhere in the video driver or something like this... in which case removing unnecessary (?) GPU reinitialization (two times on every seek vs. one time) may help a bit.

**Comment by CrendKing** [comment]
About the reload twice issue. The flush comes from upstream, twice. And I can only get this behavior in MPC-HC. MPC-BE only flushes once. I also tested with EVR and madVR on MPC-HC. Not related to renderer.

The call stack of the two flushes are exactly the same. Could be a bug of MPC-HC. You could file a ticket to clsid2.

EVR call stack
```
avisynth_filter_64.ax!AvsFilter::FrameHandler::Flush() Line 161
avisynth_filter_64.ax!AvsFilter::CAviSynthFilter::EndFlush() Line 372
avisynth_filter_64.ax!CTransformInputPin::EndFlush() Line 740
LAVVideo.ax!00007ff8142ecef1()
LAVVideo.ax!00007ff8143714f4()
LAVSplitter.ax!00007ff81dde6a7f()
LAVSplitter.ax!00007ff81dded86c()
LAVSplitter.ax!00007ff81ddedd0d()
LAVSplitter.ax!00007ff81dde7aad()
quartz.dll!CPosPassThru::SetPositions(__int64 *,unsigned long,__int64 *,unsigned long)
mpc-hc64.exe!00007ff773c48ad1()
quartz.dll!CPosPassThru::SetPositions(__int64 *,unsigned long,__int64 *,unsigned long)
quartz.dll!CFGControl::CImplMediaSeeking::SetMediaTime(__int64 *,unsigned long,__int64 *,unsigned long)
quartz.dll!CFGControl::CImplMediaSeeking::SetPositions(__int64 *,unsigned long,__int64 *,unsigned long)
mpc-hc64.exe!00007ff773a2cfee()
mpc-hc64.exe!00007ff7739f630b()
mpc-hc64.exe!00007ff773d043c2()
mpc-hc64.exe!00007ff773d05ba3()
mpc-hc64.exe!00007ff773a351f8()
mpc-hc64.exe!00007ff773cff726()
mpc-hc64.exe!00007ff773d001d0()
user32.dll!UserCallWinProcCheckWow()
user32.dll!DispatchMessageWorker()
mpc-hc64.exe!00007ff773d1f996()
mpc-hc64.exe!00007ff773d20209()
mpc-hc64.exe!00007ff773efd833()
mpc-hc64.exe!00007ff773ea5d36()
kernel32.dll!00007ff830697034()
ntdll.dll!00007ff83097cec1()
```

**Comment by chainikdn** [comment]
Indeed... ok, if there's nothing we could do about it here I'll close this for now.

**Comment by CrendKing** [comment]
FYI, tested on the released version of MPC-HC 1.9.7 x64 and the development 1.9.7.100 x64.
